### PR TITLE
Clean up BigMLClient + fix Java 8 compilation error

### PR DESCRIPTION
**Pull request by sdesimone**
This PR refactors BigMLClient implementation to remove widepsread code replication across init methods. Additionally, it fixes a compilation error when using Java 8 compiler and adds support for BIGML_USERNAME and BIGML_API_KEY environment variables (in addition to the already supported BIGML_USERNAME and BIGML_API_KEY Java properties).

### CODE REVIEW

File: src/main/java/org/bigml/binding/BigMLClient.java [hunk 1]
```diff
@@ -199,34 +199,20 @@ public static void resetInstance() {
...
-            if (this.bigmlUser == null || this.bigmlUser.equals("")
-                    || this.bigmlApiKey == null || this.bigmlApiKey.equals("")) {
-                AuthenticationException ex = new AuthenticationException(
-                        "Missing authentication information.");
-                logger.info(instance.toString(), ex);
-                throw ex;
-            }
-        }
-
-        // The seed to be used to create deterministic samples and models
```
Comment (mmerce) [inline src/main/java/org/bigml/binding/BigMLClient.java:224]: I know that's a deeper change, but the seed is not "connection" property. Is one parameter in the models configuration... and sometimes there's more than one seed, so I don't think this should be set here. We can leave that for another PR but then I'll add an issue...

**Comment by sdesimone** [comment]
Since I am going to move the Java bindings forward, I think we can leave this for the next PR.

**Comment by mmerce** [comment]
Fine. So I'll try to test.

**Comment by sdesimone** [comment]
BTW — this PR in itself has no priority…
So it could also be delayed

**Comment by mmerce** [comment]
Well, now that you are set to it, it's better to close it now that we both have the code in our minds.

**Pull request closed without merging**